server/src/main/java/org/elasticsearch/index/mapper/TextFieldMapper.java

```diff
@@ -360,7 +361,7 @@ PrefixFieldType setAnalyzer(NamedAnalyzer delegate) {
         }
 
         boolean accept(int length) {
-            return length >= minChars && length <= maxChars;
+            return length >= minChars - 1 && length <= maxChars;
```

Comment:
Let's go even further, change this to just `length <= maxChars`, and then append `minChars - prefixTerm.length` wildcards to the automaton that is used for querying?